simpleflow/execute.py

```diff
@@ -181,7 +183,21 @@ def execute(*args, **kwargs):
                     close_fds=close_fds,
                     pass_fds=pass_fds,
                 )
-                rc = process.wait()
+
+                # Wait for process
+                if timeout:
+                    t_start = time.time()
```

Comment:
+1 for Benjamin analysis, I think it's better to do it this "manually" in that case. Unless we missed something?